src/dfmt/indentation.d

```diff
@@ -278,6 +278,12 @@ private:
 
                 immutable currentIsNonWrapTemp = !details[i].wrap
                     && details[i].temp && arr[i] != tok!")" && arr[i] != tok!"!";
+
+                if (currentIsNonWrapTemp && arr[i + 1] == tok!"]")
```

Comment:
`arr[i + 1] == tok!"]"` looks super random here (and possible range violation?), can you explain what exactly it does, maybe in a comment?

otherwise the PR LGTM I think

Reply:
hm shouldn't `[` use a different indentation (temp? wrap?) instead of doing it on the `]`? What about nested arrays with this approach?

Reply:
Works for tests like:

```d
private void selectMember(Args...)(Args args)
{
    if (true)
    {
        if (args[
            [
                0,
            ],
            [
                2,
            ],
        ])
        {
        }
        else
        {
        }
    }
}
```

For nested arrays this rule doesn't apply, because a nested array is a different sequence of indentation tokens, its `]` followed by another `]`.

Not sure about implementing it as a different kind of token. The documentation for temp states:

> temporary indentation which get's reverted when a block starts generally true for all tokens except ), {, case, @

Array indentation isn't temporary, it just shouldn't be summed up with the brace indentation.

For wrap:

> generally true for all operators except {, case, @, ], (, )

Wrapping is as far as I understand more a tool to split long sequences of operators or UFCS chains into multiple lines. Array is an item list, it seems to be a bit different.